lib/auth/native/native.go

```diff
@@ -42,16 +52,46 @@ type keyPair struct {
 type nauth struct {
 	generatedKeysC chan keyPair
 	closeC         chan bool
-	closed         int32
+	mutex          sync.Mutex
+	// in production set to nil (used for tests)
+	testKey *keyPair
 }
 
+// New returns a pointer to a key generator for production purposes
 func New() *nauth {
-	n := nauth{
-		generatedKeysC: make(chan keyPair, PrecalculatedKeysNum),
-		closeC:         make(chan bool),
+	singleton.mutex.Lock()
+	defer singleton.mutex.Unlock()
+
+	if singleton.generatedKeysC == nil && PrecalculatedKeysNum > 0 {
+		singleton.generatedKeysC = make(chan keyPair, PrecalculatedKeysNum)
+		go singleton.precalculateKeys()
+	}
+	return &singleton
+}
+
+// NewTestInstance
+func SetTestKeys() {
```

Comment:
I removed all of this completely. `testauthority` is the only mocking (as before). I actually don't think it's safe to use these tricks without adopting build flags. (test code _must_ be excluded from production builds, so we can rely on build errors against accidental inclusion of something like this into production code)